src/_cffi_src/build_openssl.py

```diff
@@ -46,7 +49,16 @@ def _extra_compile_args(platform):
     When we drop support for CRYPTOGRAPHY_OPENSSL_LESS_THAN_110 we can
     revisit this.
     """
-    if platform not in ["win32", "hp-ux11", "sunos5"]:
+    # make sure the compiler used supports the flags to be added
+    is_gcc = False
+    if get_default_compiler() == "unix":
+        d = dist.Distribution()
+        cmd = config(d)
+        cmd._check_compiler()
+        is_gcc = ("gcc" in cmd.compiler.compiler[0] or
```

Comment:
This can be written as `is_gcc = cmd.compiler.compiler[0] in ["gcc", "clang"]`.

Reply:
No it can't.

Reply:
As noted in IRC: ugh, stupid string can be things like `clang-3.8` so my proposed switch won't work. C toolchains are such garbage.